addons/account_peppol/models/account_edi_proxy_user.py

```diff
@@ -494,6 +493,21 @@ def _peppol_deregister_participant(self):
         self.company_id._reset_peppol_configuration()
         self.unlink()
 
+    def _peppol_deregister_participant_to_sender(self):
+        self.ensure_one()
+
+        if self.company_id.account_peppol_proxy_state == 'receiver':
+            # fetch all documents and message statuses before unlinking the edi user
+            # so that the invoices are acknowledged
+            self._cron_peppol_get_message_status()
+            self._cron_peppol_get_new_documents()
+            if not modules.module.current_test:
+                self.env.cr.commit()
+
+        if self.company_id.account_peppol_proxy_state != 'sender':
```

Comment:
Not sure this if is really useful since the route will fail for some states (through the `accepted_states`, that are anyway not just != sender).